clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/InAppQueue.kt

```diff
@@ -0,0 +1,65 @@
+package com.clevertap.android.sdk.inapp
+
+import androidx.annotation.WorkerThread
+import com.clevertap.android.sdk.CleverTapInstanceConfig
+import com.clevertap.android.sdk.Logger
+import com.clevertap.android.sdk.inapp.store.preference.StoreRegistry
+import org.json.JSONArray
+import org.json.JSONObject
+
+class InAppQueue(
+    private val config: CleverTapInstanceConfig,
+    private val storeRegistry: StoreRegistry
+) {
+
+    @WorkerThread
+    @Synchronized
+    fun enqueue(jsonObject: JSONObject) {
+        val currentQueue = getQueue()
+        currentQueue.put(jsonObject)
+        saveQueue(currentQueue)
+    }
+
+    @WorkerThread
+    @Synchronized
+    fun enqueueAll(jsonArray: JSONArray) {
+        val currentQueue = getQueue()
+        for (i in 0 until jsonArray.length()) {
+            try {
```

Comment:
why do we use try catch here?

Reply:
try catch is more expensive at runtime compared to if check, lets add if condition and check jsonobject if possible.

Reply:
1. `jsonArray.getJSONObject(i)` throws JSONException
2. `currentQueue.put()` Throws:
UnsupportedOperationException – if the add operation is not supported by this list
ClassCastException – if the class of the specified element prevents it from being added to this list
NullPointerException – if the specified element is null and this list does not permit null elements
IllegalArgumentException – if some property of this element prevents it from being added to this list

Let's keep try catch only